SS2-Project/Assets/Starstorm2/Modules/Characters/Monsters/Mimic.cs

```diff
@@ -17,16 +23,76 @@ public sealed class Mimic : SS2Monster
+		//public string printItemList = "";
+		//
+		//[RiskOfOptionsConfigureField(SS2Config.ID_MONSTER, configDescOverride = "List of all AI Blacklisted items. Items that either don't help the mimic, or are likely to be outright unfair.")]
+		//public string printItemListAIBan = "";
+		//
+		//[RiskOfOptionsConfigureField(SS2Config.ID_MONSTER, configDescOverride = "List of items the mimic cannot use. ")]
+		//public string mimicBannedItems = ""
+
 		public override void Initialize()
 		{
 			_masterPrefab = AssetCollection.FindAsset<GameObject>("MimicMaster");
+
 			GlobalEventManager.onServerDamageDealt += ServerDamageStealItem;
-			//On.RoR2.UI.PingIndicator.RebuildPing += RebuildPingOverrideInteractable;
+			On.RoR2.UI.PingIndicator.RebuildPing += RebuildPingOverrideInteractable;
+			On.RoR2.RandomizeSplatBias.Setup += SplatSetup;
+
 			StealItemDamageType = R2API.DamageAPI.ReserveDamageType();
+
+			itemOrb = Addressables.LoadAssetAsync<GameObject>("RoR2/Base/Common/VFX/ItemTakenOrbEffect.prefab").WaitForCompletion();
+
+			itemStarburst = AssetCollection.FindAsset<GameObject>("Chest1Starburst");
+			zipperVFX = AssetCollection.FindAsset<GameObject>("ChestUnzipReal");
+
+			var pip = AssetCollection.FindAsset<GameObject>("MimicBodyNew").GetComponent<PingInfoProvider>();
+			var ping = Addressables.LoadAssetAsync<Sprite>("RoR2/Base/ChestIcon_1.png").WaitForCompletion();
+			pip.pingIconOverride = ping;
+			var mid = AssetCollection.FindAsset<InspectDef>("idMimic");
+			mid.Info.Visual = ping;
 		}
 
-		private void RebuildPingOverrideInteractable(On.RoR2.UI.PingIndicator.orig_RebuildPing orig, RoR2.UI.PingIndicator self)
+        private void SplatSetup(On.RoR2.RandomizeSplatBias.orig_Setup orig, RandomizeSplatBias self)
+        {
+			var mpc = self.GetComponent<MimicPingCorrecter>();
+            if (mpc)
+            {
+				var ml = self.GetComponent<ModelLocator>();
+                if (ml)
+                {
+					var transf = ml.modelTransform;
+					var componentsInChildren = transf.GetComponentsInChildren<Renderer>();
+					foreach(var comp in componentsInChildren)
+                    {
+						Material material2 = UnityEngine.Object.Instantiate<Material>(comp.material);
+						self.materialsList.Add(material2);
+						comp.material = material2;
+						self._propBlock = new MaterialPropertyBlock();
+						comp.GetPropertyBlock(self._propBlock);
+						self._propBlock.SetFloat("_RedChannelBias", UnityEngine.Random.Range(self.minRedBias, self.maxRedBias));
+						self._propBlock.SetFloat("_BlueChannelBias", UnityEngine.Random.Range(self.minBlueBias, self.maxBlueBias));
+						self._propBlock.SetFloat("_GreenChannelBias", UnityEngine.Random.Range(self.minGreenBias, self.maxGreenBias));
+						comp.SetPropertyBlock(self._propBlock);
+					}
+				}
+			}
+            else
+            {
+				orig(self);
+            }
+		}
+        
+		//i dont want to write IL
```

Comment:
comment explaining what this function is doing, this is an uncommon hook I imagine so why is it needed